opal/mca/accelerator/accelerator.h

```diff
@@ -110,19 +112,28 @@ struct opal_accelerator_stream_t {
     void *stream;
 };
 typedef struct opal_accelerator_stream_t opal_accelerator_stream_t;
+OBJ_CLASS_DECLARATION(opal_accelerator_stream_t);
+
+/* Constant indicating the default/zero stream */
+#define MCA_ACCELERATOR_STREAM_DEFAULT (opal_accelerator_stream_t *)0x00000002
 
 #define IPC_MAX_HANDLE_SIZE 64
 struct opal_accelerator_ipc_handle_t {
+    opal_object_t super;
     size_t size;
     uint8_t handle[IPC_MAX_HANDLE_SIZE];
+    void* dev_ptr;
```

Comment:
after opening an IPC handle, the function returns a device pointer that can contains the address under which the remote memory location be can accessed. We need to store this address since it is required during the CloseMemhandle function as an argument for both CUDA and ROCM. 

```
CUresult cuIpcCloseMemHandle (Cudeviceptr dptr );
hipError_t hipIpcCloseMemHandle (void *devPtr);
```

The destructor interfaces only provide the opal_accelerator_ipc_handle_t structure that we 'destruct', hence the pointer has to be stored as part of the handle.